tests/org.dataflowanalysis.analysis.converter.tests/src/org/dataflowanalysis/analysis/converter/tests/ConverterTests.java

```diff
@@ -0,0 +1,303 @@
+		
+		final var usageModelPath = Paths.get("models", name, modelFileName + ".usagemodel").toString();
+		final var allocationPath = Paths.get("models", name, modelFileName + ".allocation").toString();
+		final var nodeCharPath = Paths.get("models", name, modelFileName + ".nodecharacteristics").toString();
+				
+		DataFlowConfidentialityAnalysis analysis = new PCMDataFlowConfidentialityAnalysisBuilder()
+		        .standalone()
+		        .modelProjectName(TEST_MODEL_PROJECT_NAME)
+		        .usePluginActivator(Activator.class)
+		        .useUsageModel(usageModelPath)
+		        .useAllocationModel(allocationPath)
+		        .useNodeCharacteristicsModel(nodeCharPath)
+		        .build();
+		
+		analysis.initializeAnalysis();
+		analysis.findAllSequences();
+		var sequences = analysis.findAllSequences();
+		var propagationResult = analysis.evaluateDataFlows(sequences);
+		
+		Map<String,String> assIdToName = new HashMap<>();
+		for(ActionSequence as : propagationResult) {
+			for(AbstractActionSequenceElement ase: as.getElements()) {
+				if(ase instanceof SEFFActionSequenceElement) {
+					var cast=(SEFFActionSequenceElement) ase;
+					assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+				}
+				else if(ase instanceof CallingSEFFActionSequenceElement) {
+					var cast=(CallingSEFFActionSequenceElement) ase;
+					assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+				}
+				else if(ase instanceof CallingUserActionSequenceElement) {
+					var cast=(CallingUserActionSequenceElement) ase;
+					assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+				}
+				else {
+					var cast=(UserActionSequenceElement) ase;
+					assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+				}
+			}
+		}
+		
+		ProcessASS ass2dfd = new ProcessASS();
+		
+		ass2dfd.transform(propagationResult);
+		
+		DataDictionary dd = ass2dfd.getDictionary();
+		DataFlowDiagram dfd = ass2dfd.getDataFlowDiagram();
+		
+		assertEquals(dfd.getNodes().size(),assIdToName.keySet().size());
+		
+		List<String> nodeIds = new ArrayList<>();
+		for(Node node :dfd.getNodes()) {
+			nodeIds.add(node.getId());
+		}
+		Collections.sort(nodeIds);
+		List<String> assIds=new ArrayList<>(assIdToName.keySet());
+		Collections.sort(assIds);
+		
+		assertEquals(assIds,nodeIds);
+		
+		for(Node node : dfd.getNodes()) {
+			assertEquals(node.getEntityName(),assIdToName.get(node.getId()));
+		}
+
+		List<String> flowNames=new ArrayList<>();
+		for(ActionSequence as : propagationResult) {
+			for(AbstractActionSequenceElement ase: as.getElements()) {
+				List<DataFlowVariable> variables=ase.getAllDataFlowVariables();
+				for(DataFlowVariable variable:variables) {
+					flowNames.add(variable.variableName());
+				}
+			}
+		}
+
+		assertEquals(flowNames.size(),dfd.getFlows().size());
+		
+	}
+	
+	public static void cleanup(String path) {
```

Comment:
The tests should make assertions on the actual output of the converters without saving the files, this would remove the storing and deleting logic from the actual transformation logic.